api/src/main/java/org/pmiops/workbench/api/ClusterController.java

```diff
@@ -116,7 +124,65 @@
   }
 
   @Override
-  public ResponseEntity<ClusterListResponse> listClusters(
+  @AuthorityRequired(Authority.SECURITY_ADMIN)
+  public ResponseEntity<List<ListClusterResponse>> deleteClustersInProject(
+      String billingProjectId, ListClusterDeleteRequest clusterNamesToDelete) {
+    if (billingProjectId == null) {
+      throw new BadRequestException("Must specify billing project");
+    }
+    List<org.pmiops.workbench.notebooks.model.ListClusterResponse> clustersToDelete =
+        leonardoNotebooksClient.listClustersByProjectAsAdmin(billingProjectId).stream()
+            .filter(
+                cluster ->
+                    clusterNamesToDelete.getClustersToDelete() == null
+                        || clusterNamesToDelete
+                            .getClustersToDelete()
+                            .contains(cluster.getClusterName()))
+            .collect(Collectors.toList());
+
+    clustersToDelete.forEach(
+        cluster ->
+            leonardoNotebooksClient.deleteClusterAsAdmin(
+                cluster.getGoogleProject(), cluster.getClusterName()));
+    List<ListClusterResponse> clustersInProjectAffected =
+        leonardoNotebooksClient.listClustersByProjectAsAdmin(billingProjectId).stream()
```

Comment:
Hmmm... I don't think we want to list all deleted clusters though, because we reuse names. We may be okay with this, because clusters probably don't delete that quickly? It will affect return value if they do delete quickly, but won't affect auditing.

Reply:
agreed offline - we do not want to list deleted